### PR TITLE
Fix

### PR DESCRIPTION
**Pull request by slizko1**

### CODE REVIEW

**Review by Bohadan-Markatov — APPROVED** [review]

File: src/main/java/core/basesyntax/service/RegistrationServiceImpl.java [hunk 1]
```diff
@@ -5,10 +5,54 @@
...
+        passwordNullCheck(user);
+        ageCheck(user);
+        userRegistrationCheck(user);
+        loginLengthCheck(user);
+        passwordLengthCheck(user);
+        return storageDao.add(user);
+    }
+
+    private void passwordLengthCheck(User user) {
+        if (user.getPassword().length() < 6) {
```
Comment (karma-o) [inline src/main/java/core/basesyntax/service/RegistrationServiceImpl.java:23]: make 6 a constant

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
```
Comment (karma-o) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:9]: missing test cases:
* existing user with the same login(keep in mind the way how we should properly add the first user, check common mistakes for it)
* null user itself

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    private static final int VALID_AGE = 20;
+    private static final int INVALID_AGE = 17;
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private final User user = new User();
+
+    @Test
+    public void registerValidUser_Ok() {
+        user.setLogin(VALID_LOGIN);
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
```
Comment (karma-o) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:23]: let's not duplicate the `user` initialization, and bring it out to the `beforeEach`, where we should also reinitialize it with `new User()`, and let's rename this var to a `validUser` and it the test itself we can then leave just  single invalid field assignment

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceImplTest {
+    private static final String VALID_LOGIN = "user123";
+    private static final String INVALID_LOGIN = null;
+    private static final String VALID_PASSWORD = "password123";
+    private static final String INVALID_PASSWORD = "nvdf";
+    private static final int VALID_AGE = 20;
```
Comment (karma-o) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:14]: also better to test with edge cases
```suggestion
    private static final int VALID_AGE = 18;
```

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    @Test
+    public void registerUserWithNullLoginThrowRegistrationException_notOk() {
+        user.setLogin(INVALID_LOGIN); // Use the constant for null/empty login
+        user.setPassword(VALID_PASSWORD);
+        user.setAge(VALID_AGE);
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(user));
+        assertEquals("Login can't be null", exception.getMessage()); // Check exception message
```
Comment (karma-o) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:35]: let's do that for other tests as well

**Review by karma-o — CHANGES_REQUESTED** [review]
Looks good so far, just some minor tings left

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+    @Test
+    public void registerValidUser_Ok() {
+        User registeredUser = registrationService.register(validUser);
+        assertEquals(validUser, registeredUser);
+    }
+
+    @Test
+    void registerAlreadyAddedUserThrowRegistrationException_notOk() {
+        User registeredUser = registrationService.register(validUser);
```
Comment (karma-o) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:49]: check common mistakes, we shouldn't use `register()` for adding the first user to the storage, because this method is under a test, and we can not guarantee that it is working properly, thus it can give us a false-positive result. Instead we should use `storage` or `storageDao`

File: src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+        RegistrationException exception = assertThrows(RegistrationException.class,
+                () -> registrationService.register(registeredUser));
+        assertEquals("User - " + registeredUser.getLogin()
+                + " is already registered", exception.getMessage());
```
Comment (karma-o) [inline src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java:53]: I'd rather extract this first parameter to some `expectedMessage` variable to improve readability

**Review by karma-o — APPROVED** [review]
Approved, but please check the comments regarding the common mistake
